src/app/shared-services/trajectory-import-export.service.ts

```diff
@@ -0,0 +1,281 @@
+import { Injectable } from '@angular/core'
+import { Trajectory, TrajectoryMeta, TrajectoryType } from '../model/trajectory'
+import { TrajectoryService } from '../shared-services/trajectory.service'
+import { v4 as uuid } from 'uuid'
+import {
+  ActionSheetController,
+  LoadingController,
+  Platform,
+  ToastController,
+} from '@ionic/angular'
+import { HttpClient } from '@angular/common/http'
+import { SqliteService } from './db/sqlite.service'
+import { LocationService } from './location.service'
+import {
+  Plugins,
+  FilesystemDirectory,
+  FilesystemEncoding,
+} from '@capacitor/core'
+import { SocialSharing } from '@ionic-native/social-sharing/ngx'
+import { AndroidPermissions } from '@ionic-native/android-permissions/ngx'
+const { FileSelector, Filesystem } = Plugins
+
+@Injectable()
+export class TrajectoryImportExportService extends TrajectoryService {
```

Comment:
I guess the Angular-way is not to inherit, but to make it a member. Only `TrajectoryService.addTrajectory()` is used right?
But you probably can leave it as is

Reply:
Using `TrajectoryService.addTrajectory()` as well as `TrajectoryService.getOne()`.

Since this service gives a somewhat extended access to reading/writing trajectories from/to the persistent, Ifor me it makes sense to treat it as an actual extension for `TrajectoryService` - but I'm open to discussions :)